homeassistant/components/statistics/sensor.py

```diff
@@ -598,71 +608,67 @@ def _callable_characteristic_fn(
     # Statistics for numeric sensor
 
     def _stat_average_linear(self) -> StateType:
-        if len(self.states) >= 2:
-            area: float = 0
-            for i in range(1, len(self.states)):
-                area += (
-                    0.5
-                    * (self.states[i] + self.states[i - 1])
-                    * (self.ages[i] - self.ages[i - 1]).total_seconds()
-                )
-            age_range_seconds = (self.ages[-1] - self.ages[0]).total_seconds()
+        if (states := self.states) and len(states) >= 2:
+            ages = self.ages
```

Comment:
Same as above. This addition does not make sense to me. Happy to learn about your reasoning

Reply:
The self.ages lookup isn't free by storing in a local variable it avoid the getattr under the hood each call

Reply:
I understand what you are saying. I did not expect any performance difference from it (vs the cost of the local variable). Would you by any chance know an article talking about this?

I won't stand in the way of performance improvements, it is ... just... so ugly 😭☺️

Reply:
https://wiki.python.org/moin/PythonSpeed
> In functions, local variables are accessed more quickly than global variables, builtins, and attribute lookups. So, it is sometimes worth localizing variable access in inner-loops. For example, the code for random.shuffle() localizes access with the line, random=self.random. That saves the shuffling loop from having to repeatedly lookup self.random. Outside of loops, the gain is minimal and rarely worth it.